### PR TITLE
WIP: improve flash directions for redboard

### PR DESCRIPTION
**Pull request by ppannuto**
### Pull Request Overview

This adds documentation and improves the redboard flashing scripts a bit.

### Testing Strategy

WIP: I still haven't been able to flash from my machine, but getting closer....

### TODO or Help Wanted

```
$ PORT=/dev/tty.usbserial-14412430 make flash
    Finished release [optimized + debuginfo] target(s) in 0.19s
   text	   data	    bss	    dec	    hex	filename
  73729	   1172	   7016	  81917	  13ffd	/Volumes/code/helena-project/tock/target/thumbv7em-none-eabi/release/redboard_artemis_nano
10f0935063cb56101bb9e184f64ca7b6e079a14cf4462662d64ef814dff6355c  /Volumes/code/helena-project/tock/target/thumbv7em-none-eabi/release/redboard_artemis_nano.bin
ambiq/ambiq_bin2board.py --bin /Volumes/code/helena-project/tock/target/thumbv7em-none-eabi/release/redboard_artemis_nano.bin --load-address-blob 0x20000 -b 115200 -port /dev/tty.usbserial-14412430 -r 2 -v --magic-num 0xCB --version 0x0 --load-address-wired 0xc000 -i 6 --options 0x1
Header Size =  0x80
original app_size  0x12494 ( 74900 )
load_address  0xc000 ( 49152 )
app_size  0x12494 ( 74900 )
w0 = 0xcb012514
Security Value  0x10
w2 =  0x10008080
addrWord =  0xc000
versionKeyWord =  0x0
child0/feature =  0xffffffff
child1 =  0xffffffff
crc =   0x212c255
Writing to file  wuimage_OTA_blob.bin
testing: wuimage_OTA_blob.bin
Header Size =  0x60
app_size  0x12514 ( 75028 )
Writing to file  wuimage_Wired_OTA_blob.bin
Image from  0x0  to  0x12514  will be loaded at 0x20000
Connecting over serial port /dev/tty.usbserial-14412430...
Sending Hello.
No response for command 0x00000000
received bytes 1
['0xfc']
Failed to respond
Fail
Sending Hello.
No response for command 0x00000000
received bytes 1
['0xfc']
Failed to respond
Fail
Sending Hello.
No response for command 0x00000000
received bytes 1
['0xfe']
Failed to respond
Fail
Tries = 3
Upload failed
```

I'm hung up here currently, @alistair23, do I need to do anything special the first time I'm trying to flash to overwrite the old bootloader? I have verified that's the correct serial port.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [ ] Ran `make prepush`.

### CODE REVIEW

File: boards/redboard_artemis_nano/Makefile [hunk 1]
```diff
@@ -11,8 +11,8 @@ endif
 
 .PHONY: flash
 flash: $(TOCK_ROOT_DIRECTORY)target/$(TARGET)/release/$(PLATFORM).bin
-	python ambiq/ambiq_bin2board.py --bin $< --load-address-blob 0x20000 -b 115200 -port $(PORT) -r 2 -v --magic-num 0xCB --version 0x0 --load-address-wired 0xc000 -i 6 --options 0x1
+	ambiq/ambiq_bin2board.py --bin $< --load-address-blob 0x20000 -b 115200 -port $(PORT) -r 2 -v --magic-num 0xCB --version 0x0 --load-address-wired 0xc000 -i 6 --options 0x1
```
Comment (alistair23) [inline boards/redboard_artemis_nano/Makefile:14]: Why does this help?

Does changing this to `python3` instead of `python` also work?

Comment (ppannuto) [reply boards/redboard_artemis_nano/Makefile:14]: It would, but the script has a shebang at the beginning, so I figure we may as well use it.

Comment (alistair23) [reply boards/redboard_artemis_nano/Makefile:14]: But then you had to change it to be executable, which is a difference from the original. Not a big deal, but still something that we don't really need.

**Comment by bradjc** [comment]
Should we merge these changes at least?

**Comment by alistair23** [comment]
I would say instead that using `python3` instead of `python` would be the better approach.

**Pull request closed without merging**